tests/test_text.py

```diff
@@ -29,12 +29,16 @@ def test_parse_yaml():
     assert set(o) == {"a", "b"}
 
 
-@pytest.mark.parametrize("yaml_url, expected", [("https://raw.githubusercontent.com/GeoscienceAustralia/dea-config/"\
-                                      "master/dev/services/odc-stats/fc_percentile/ga_ls_fc_pc_cyear_3.yaml",
-                                      "ga_ls_fc_pc_cyear_3")])
-def test_load_yaml_remote(yaml_url, expected):
-    content = load_yaml_remote(yaml_url)
-    assert content["product"]["name"] == expected
+def test_load_yaml_remote(httpserver):
+    httpserver.expect_request("/ga_ls_fc_pc_cyear_3.yaml").respond_with_data("""
+    a: 3
+    b: foo
+    """
+    )
+    content = load_yaml_remote(httpserver.url_for("/ga_ls_fc_pc_cyear_3.yaml"))
+    assert content["a"] == 3 and content["b"] == "foo"
+    with pytest.raises(FileNotFoundError, match=r".*something_non_exist.yaml"):
+        load_yaml_remote(httpserver.url_for("/something_non_exist.yaml"))
```

Comment:
I think you should change this to use a fake URL, rather than `httpserver.url_for`, as you're catching the error of the library you're using to do the test, rather than the underlying function.

So this test is not actually helpful!

Reply:
the scenario is that the server is alive, but the file doesn't exist on the server since the server doesn't respond to the request of `/something_non_exist.yaml`. Is it the fundamentally different than the server does not exist with a arbitrary url?

Reply:
If you do:

```
with fsspec.open("http://example.com/fake.yaml", mode="r") as f:
    y = next(yaml.safe_load_all(f))
```

it raises `ClientResponseError`.

Reply:
so that's `fsspec` version related behaviour, not url related.
If `fsspec >= 2021.11.*`, the error is
```
FileNotFoundError: http://example.com/fake.yaml
```
if `fsspec` is much older, the error is
```
ClientResponseError
```
for both cases.

That's my latest commit trying to address, in bumping `fsspec` version in both test and install

Reply:
Ok, I see. Thanks!